sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -50,6 +50,8 @@ trait InvokeLike extends Expression with NonSQLExpression with ImplicitCastInput
 
   def propagateNull: Boolean
 
+  override def foldable: Boolean =
+    children.forall(_.foldable) && !dataType.isInstanceOf[ObjectType]
```

Comment:
Actually  .. shouldn't we set deterministic as `false` since the function can be indeterministic like UDF?

Reply:
> Actually .. shouldn't we set deterministic as `false` since the function can be indeterministic like UDF?

Should we add a new parameter to define if the `InvokeLike` can be deterministic?  like nullable?

Reply:
Good catch and it's a bug we must fix first. It turns out that we never respect `BoundFunction.isDeterministic` which can cause many issues when doing project collpase, filter pushdown, etc. and leads to wrong query result.

I think we need to add a `deterministic` flag into `InvokeLike` and set it when use it to call DS v2 funciton. @sunchao can you help to fix this issue?

also cc @huaxingao as ds v2 function was released in 3.2.0 and we might want to fix it in 3.2.1.

Reply:
Should wait for  add `deterministic` finished then this pr can be merged safety.

Reply:
Thanks, let me fix that.

Reply:
opened #35243 for this